### ISSUE TITLE
The order in which SVGO plugins are run should not be determined by the order in which the plugins are listed in svgo.json

### ISSUE THREAD

**Issue by alexjlockwood** [BohemianCoding/svgo-compressor#41]
Currently the order in which SVGO plugins are executed on an exported SVG depends on the order in which the plugins are specified in the `svgo.json` configuration file: https://github.com/BohemianCoding/svgo-compressor/blob/18d341102a6533bdc292a3962e81ecc4ff92aa18/src/plugin.js#L37-L86

The plugin execution order in SVGO is fixed in order to guarantee the highest possible optimized result. You can see the order in which SVGO executes plugins here: https://github.com/svg/svgo/blob/master/.svgo.yml. After parsing the plugins from the `svgo.json` config file, this Sketch plugin should sort the enabled plugins according to this predefined order before executing svgo on the exported SVG.

The reason why the order of SVGO plugins is important is discussed a bit here . (specifically the part where it mentions that every plugin "has a specific position in the plugins list"): https://github.com/svg/svgo/blob/ba07843565aa50cd9dfde0c8806921c14f532096/docs/how-it-works/en.md

Basically, SVGO plugins are designed to build on each other as they are executed one by one. For example, consider the below SVG:

```xml
<svg>
  <path d="M 0 0 L 100 100" />
  <g>
    <path d="M 200 200 L 300 300" />
  </g>
</svg>
```

Executing the `mergePaths` plugin before the `collapseGroups` plugin will result in the following SVG (first merge sibling paths, and then remove unnecessary groups):

```xml
<svg>
  <path d="M 0 0 L 100 100" />
  <path d="M 200 200 L 300 300" />
</svg>
```

Whereas executing `collapseGroups` before `mergePaths` will give much better results (first remove unnecessary groups, which increases the chances that there will be sibling paths that can be merged):

```xml
<svg>
  <path d="M 0 0 L 100 100 M 200 200 L 300 300" />
</svg>
```

**Comment by mathieudutour** [comment]
IIRC the order of the default config of the plugin is the same as https://github.com/svg/svgo/blob/master/.svgo.yml exactly for the reasons you mentioned. If it's not, then it's a bug and should be fixed!

I don't think there is a reason for the sketch plugin to force an order for the svgo plugins (you can also change the order of the plugins in svgo if you want to). If someone chooses to change it, there might be a good reason for it. Do you believe otherwise?

**Comment by alexjlockwood** [comment]
I agree. I think the default ordering used to be alphabetical but it seems like that must have been changed recently. :)
